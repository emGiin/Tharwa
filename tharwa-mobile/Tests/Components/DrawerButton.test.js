import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import DrawerButton from '../../App/Components/DrawerButton'

describe('DrawerButton component', () => {
  let container, content;
  const spy = jest.fn();

  beforeAll(() => {
    container = shallow(<DrawerButton onPress={spy} text='test' />)
    content = container.dive()
  })

  it('should render correctly', () => {
    expect(content).toMatchSnapshot()
  })

  it('should call on press function', () => {
    container.simulate('press')
    expect(spy).toHaveBeenCalled()
  })
})