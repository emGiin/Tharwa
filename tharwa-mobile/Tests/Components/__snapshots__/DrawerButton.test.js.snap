// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawerButton component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TouchableOpacity
    activeOpacity={0.2}
    onPress={[MockFunction]}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontFamily": "Avenir-Book",
          "fontSize": 20,
          "marginVertical": 10,
        }
      }
    >
      test
    </Text>
  </TouchableOpacity>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "accessibilityComponentType": undefined,
      "accessibilityLabel": undefined,
      "accessibilityTraits": undefined,
      "accessible": true,
      "children": Array [
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontFamily": "Avenir-Book",
              "fontSize": 20,
              "marginVertical": 10,
            }
          }
        >
          test
        </Text>,
        null,
      ],
      "hitSlop": undefined,
      "isTVSelectable": true,
      "nativeID": undefined,
      "onLayout": undefined,
      "onResponderGrant": [Function],
      "onResponderMove": [Function],
      "onResponderRelease": [Function],
      "onResponderTerminate": [Function],
      "onResponderTerminationRequest": [Function],
      "onStartShouldSetResponder": [Function],
      "style": Array [
        undefined,
        Object {
          "opacity": 1,
        },
      ],
      "testID": undefined,
      "tvParallaxProperties": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "test",
          "ellipsizeMode": "tail",
          "style": Object {
            "color": "white",
            "fontFamily": "Avenir-Book",
            "fontSize": 20,
            "marginVertical": 10,
          },
        },
        "ref": null,
        "rendered": "test",
        "type": [Function],
      },
      null,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "accessibilityComponentType": undefined,
        "accessibilityLabel": undefined,
        "accessibilityTraits": undefined,
        "accessible": true,
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "fontFamily": "Avenir-Book",
                "fontSize": 20,
                "marginVertical": 10,
              }
            }
          >
            test
          </Text>,
          null,
        ],
        "hitSlop": undefined,
        "isTVSelectable": true,
        "nativeID": undefined,
        "onLayout": undefined,
        "onResponderGrant": [Function],
        "onResponderMove": [Function],
        "onResponderRelease": [Function],
        "onResponderTerminate": [Function],
        "onResponderTerminationRequest": [Function],
        "onStartShouldSetResponder": [Function],
        "style": Array [
          undefined,
          Object {
            "opacity": 1,
          },
        ],
        "testID": undefined,
        "tvParallaxProperties": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "test",
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "white",
              "fontFamily": "Avenir-Book",
              "fontSize": 20,
              "marginVertical": 10,
            },
          },
          "ref": null,
          "rendered": "test",
          "type": [Function],
        },
        null,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
